ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,264 @@
+
+
+def _get_hexidecimal_span_id(span_id: int) -> str:
+    return "{:016x}".format(span_id)
+
+
+def _get_time_from_ns(nanoseconds: int) -> Dict:
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining
+    nanoseconds"""
+    if not nanoseconds:
+        return None
+    seconds, nanos = divmod(nanoseconds, 1e9)
+    return {"seconds": int(seconds), "nanos": int(nanos)}
+
+
+def _get_truncatable_str_object(
+    str_to_convert: str, max_length: int = MAX_LENGTH
+):
+    """Truncate the string if it exceeds the length limit and record the
+    truncated bytes count."""
+    truncated, truncated_byte_count = _truncate_str(str_to_convert, max_length)
+
+    return TruncatableString(
+        value=truncated, truncated_byte_count=truncated_byte_count
+    )
+
+
+def _truncate_str(
+    str_to_check: str, limit: int = MAX_LENGTH
+) -> Tuple[str, int]:
+    """Check the length of a string. If exceeds limit, then truncate it."""
+    str_bytes = str_to_check.encode("utf-8")
+    str_len = len(str_bytes)
+    truncated_byte_count = 0
+
+    if str_len > limit:
+        truncated_byte_count = str_len - limit
+        str_bytes = str_bytes[:limit]
+
+    result = str(str_bytes.decode("utf-8", errors="ignore"))
+
+    return result, truncated_byte_count
+
+
+def _extract_status(status: trace_api.Status) -> Status:
+    """Convert a Status object to protobuf object."""
+    if not status:
+        return None
+    status_dict = {"details": None, "code": status.canonical_code.value}
+
+    if status.description is not None:
+        status_dict["message"] = status.description
+
+    return Status(**status_dict)
+
+
+def _extract_links(links: Sequence[trace_api.Link]) -> ProtoSpan.Links:
+    """Convert span.links"""
+    if not links:
+        return None
+    extracted_links = []
+    for link in links:
+        trace_id = _get_hexidecimal_trace_id(link.context.trace_id)
+        span_id = _get_hexidecimal_span_id(link.context.span_id)
+        extracted_links.append(
+            {
+                "trace_id": trace_id,
+                "span_id": span_id,
+                "type": "CHILD_LINKED_SPAN",
+                "attributes": _extract_attributes(link.attributes),
+            }
+        )
+    return ProtoSpan.Links(link=extracted_links, dropped_links_count=0)
+
+
+def _extract_events(events: Sequence[Event]) -> ProtoSpan.TimeEvents:
+    """Convert span.events to dict."""
+    if not events:
+        return None
+    logs = []
+    for event in events:
+        logs.append(
+            {
+                "time": _get_time_from_ns(event.timestamp),
+                "annotation": {
+                    "description": _get_truncatable_str_object(
+                        event.name, 256
+                    ),
+                    "attributes": _extract_attributes(event.attributes),
+                },
+            }
+        )
+    return ProtoSpan.TimeEvents(time_event=logs, dropped_annotations_count=0)
```

Comment:
`0` is already the default value for `dropped_annotations_count`. If you want to explicitly specify the value for readability reasons, you should probably also specify `dropped_message_events_count`.